modules/@shopify/checkout-sheet-kit/android/src/main/java/com/shopify/reactnative/checkoutsheetkit/RCTCheckoutWebView.java

```diff
@@ -0,0 +1,356 @@
+*/
+
+package com.shopify.reactnative.checkoutsheetkit;
+
+import android.content.Context;
+import android.net.Uri;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.AttributeSet;
+import android.util.Log;
+import android.widget.FrameLayout;
+
+import com.facebook.react.bridge.Arguments;
+import com.facebook.react.bridge.ReactContext;
+import com.facebook.react.bridge.WritableMap;
+import com.facebook.react.uimanager.events.RCTEventEmitter;
+
+import com.shopify.checkoutsheetkit.CheckoutAddressChangeRequestedEvent;
+import com.shopify.checkoutsheetkit.CheckoutException;
+import com.shopify.checkoutsheetkit.CheckoutOptions;
+import com.shopify.checkoutsheetkit.CheckoutWebView;
+import com.shopify.checkoutsheetkit.CheckoutWebViewEventProcessor;
+import com.shopify.checkoutsheetkit.DefaultCheckoutEventProcessor;
+import com.shopify.checkoutsheetkit.lifecycleevents.CheckoutCompleteEvent;
+import com.shopify.checkoutsheetkit.pixelevents.PixelEvent;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.Map;
+
+public class RCTCheckoutWebView extends FrameLayout {
+    private static final String TAG = "RCTCheckoutWebView";
+
+    private CheckoutWebView checkoutWebView;
+    private String checkoutUrl;
+    private String auth;
+    private boolean pendingSetup = false;
+    private CheckoutConfiguration lastConfiguration = null;
+    private final ObjectMapper mapper = new ObjectMapper();
+    private final Handler mainHandler = new Handler(Looper.getMainLooper());
+
+    private static class CheckoutConfiguration {
+        final String url;
+        final String authToken;
+
+        CheckoutConfiguration(String url, String authToken) {
+            this.url = url;
+            this.authToken = authToken;
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (!(obj instanceof CheckoutConfiguration)) {
+                return false;
+            }
+            CheckoutConfiguration other = (CheckoutConfiguration) obj;
+            return url.equals(other.url) &&
+                   ((authToken == null && other.authToken == null) ||
+                    (authToken != null && authToken.equals(other.authToken)));
+        }
+    }
+
+    public RCTCheckoutWebView(Context context) {
+        super(context);
+        Log.d(TAG, "RCTCheckoutWebView constructor called");
+        init();
+    }
+
+    public RCTCheckoutWebView(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        Log.d(TAG, "RCTCheckoutWebView constructor with attrs called");
+        init();
+    }
+
+    private void init() {
+        // Empty initialization - WebView will be created when URL is set
+        Log.d(TAG, "RCTCheckoutWebView initialized");
+    }
+
+    public void setCheckoutUrl(String url) {
+        if (url == null ? checkoutUrl == null : url.equals(checkoutUrl)) {
```

Comment:
```suggestion
        if (url == null || Objects.equals(url, checkoutUrl)) {
```

reads a bit more cleanly to me. thoughts?